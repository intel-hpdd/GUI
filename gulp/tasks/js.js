//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
