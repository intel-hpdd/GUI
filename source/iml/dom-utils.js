// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
