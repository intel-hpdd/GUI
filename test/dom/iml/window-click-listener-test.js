// @flow

import WindowClickListener from "../../../source/iml/window-click-listener.js";
import Inferno from "inferno";
import { renderToSnapshot } from "../../test-utils.js";
import { querySelector } from "../../../source/iml/dom-utils.js";

describe("WindowClickListener DOM testing", () => {
  let root, component, sibling;

  beforeEach(() => {
    const MockComponent = (props: { isOpen?: boolean, toggleOpen?: Function }) => {
      return (
        <span class="clicker" onClick={props.toggleOpen}>
          {props.isOpen ? "open" : "closed"}
        </span>
      );
    };

    root = document.createElement("div");
    sibling = document.createElement("div");
    querySelector(document, "body").appendChild(root);
    querySelector(document, "body").appendChild(sibling);
    component = (
      <WindowClickListener>
        <MockComponent />
      </WindowClickListener>
    );
  });

  afterEach(() => {
    querySelector(document, "body").removeChild(root);
    querySelector(document, "body").removeChild(sibling);
  });

  it("should throw on more than one child", () => {
    expect(() =>
      Inferno.render(
        <WindowClickListener>
          <div />
          <div />
        </WindowClickListener>,
        root
      )
    ).toThrow();
  });

  it("should throw on zero children", () => {
    // $FlowFixMe: This is passing wrong types on purpose
    expect(() => Inferno.render(<WindowClickListener />, root)).toThrow();
  });

  it("should render the child", () => {
    Inferno.render(component, root);
    expect(renderToSnapshot(component)).toMatchSnapshot();
  });

  it("should open when clicked", () => {
    Inferno.render(component, root);
    querySelector(root, ".clicker").click();

    expect(root.innerHTML).toMatchSnapshot();
  });

  it("should close when clicked twice", () => {
    Inferno.render(component, root);
    const clicker = querySelector(root, ".clicker");
    clicker.click();
    clicker.click();
    expect(root.innerHTML).toMatchSnapshot();
  });

  it("should close when an adjacent el is clicked", () => {
    Inferno.render(component, root);
    querySelector(root, ".clicker").click();
    sibling.click();
    expect(root.innerHTML).toMatchSnapshot();
  });
});
