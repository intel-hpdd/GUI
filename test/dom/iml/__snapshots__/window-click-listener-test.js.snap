// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowClickListener DOM testing should close when an adjacent el is clicked 1`] = `"<span class=\\"clicker\\">closed</span>"`;

exports[`WindowClickListener DOM testing should close when clicked twice 1`] = `"<span class=\\"clicker\\">closed</span>"`;

exports[`WindowClickListener DOM testing should open when clicked 1`] = `"<span class=\\"clicker\\">open</span>"`;

exports[`WindowClickListener DOM testing should render the child 1`] = `
<div>
  <span
    class="clicker"
  >
    closed
  </span>
</div>
`;
